server/tournaments/index.js

```diff
@@ -9,42 +9,51 @@ const TOURBAN_DURATION = 14 * 24 * 60 * 60 * 1000;
 
 Punishments.roomPunishmentTypes.set('TOURBAN', 'banned from tournaments');
 
+/** @type {{[k: string]: Object}} */
 let TournamentGenerators = Object.create(null);
-let generatorFiles = {
+/** @type {{roundrobin: string, elimination: string}} */
+const generatorFiles = {
 	'roundrobin': 'generator-round-robin',
 	'elimination': 'generator-elimination',
 };
 for (const type in generatorFiles) {
+	// @ts-ignore
 	TournamentGenerators[type] = require('./' + generatorFiles[type]);
 }
 
+/** @type {{[k: string]: Tournament}} */
 exports.tournaments = {};
 
+/** @param {User[]} users */
 function usersToNames(users) {
 	return users.map(user => user.name);
 }
 
-class Tournament {
+class Tournament extends Rooms.RoomGame {
+	/**
+	 * @param {ChatRoom} room
+	 * @param {Format} format
+	 * @param {Object} generator TODO: type this properly
+	 * @param {string | undefined} playerCap
+	 * @param {boolean} isRated
+	 */
 	constructor(room, format, generator, playerCap, isRated) {
-		if (room.tourNumber) {
-			room.tourNumber++;
-		} else {
-			room.tourNumber = 1;
-		}
-		format = toId(format);
+		super(room);
+		const formatId = toId(format);
 
 		this.id = room.id;
 		this.room = room;
-		this.title = Dex.getFormat(format).name + ' tournament';
+		this.title = format.name + ' tournament';
 		this.isTournament = true;
 		this.allowRenames = false;
 		this.players = Object.create(null);
 		this.playerCount = 0;
-		this.playerCap = parseInt(playerCap) || Config.tourdefaultplayercap || 0;
+		this.playerCap = playerCap ? parseInt(playerCap) : Config.tourdefaultplayercap || 0;
```

Comment:
This doesn't change my reaction.

?

What's going on here?

Reply:
the conditional is because typescript doesn't like ``parse[Int | Float](undefined)``

Reply:
```suggestion
		this.playerCap = (playerCap ? parseInt(playerCap) : Config.tourdefaultplayercap) || 0;
```

Both need to fall back, though.

Reply:
my bad, forgot how precedence worked